File: src/main/java/io/wollinger/snipsniper/scviewer/SCViewerListener.java
package io.wollinger.snipsniper.scviewer;

import io.wollinger.snipsniper.snipscope.SnipScopeListener;
import io.wollinger.snipsniper.utils.InputContainer;

import java.awt.event.KeyEvent;
import java.awt.event.MouseEvent;

public class SCViewerListener extends SnipScopeListener {
    private final SCViewerWindow scViewerWindow;
    private final InputContainer input;

    public SCViewerListener(SCViewerWindow snipScopeWindow) {
        super(snipScopeWindow);
        scViewerWindow = snipScopeWindow;
        input = scViewerWindow.getInputContainer();
    }

    @Override
    public void mousePressed(MouseEvent mouseEvent) {
        super.mousePressed(mouseEvent);
        if(mouseEvent.getButton() == 3)
            scViewerWindow.dispose();
    }

    @Override
    public void keyPressed(KeyEvent keyEvent) {
        super.keyPressed(keyEvent);
        switch (keyEvent.getKeyCode()) {
            case KeyEvent.VK_LEFT: scViewerWindow.slideImage(-1); break;
            case KeyEvent.VK_RIGHT: scViewerWindow.slideImage(1); break;
            case KeyEvent.VK_ENTER: scViewerWindow.openEditor(); break;
            case KeyEvent.VK_F5: scViewerWindow.refreshFolder(); break;
        }
    }
}

File: src/main/kotlin/net/snipsniper/scviewer/SCViewerListener.kt
package net.snipsniper.scviewer

import net.snipsniper.snipscope.SnipScopeListener
import net.snipsniper.utils.SnipFileChooser
import java.awt.event.KeyEvent
import java.awt.event.MouseEvent

class SCViewerListener(private val scViewerWindow: SCViewerWindow): SnipScopeListener(scViewerWindow) {
    override fun mousePressed(mouseEvent: MouseEvent) {
        super.mousePressed(mouseEvent)
        if(!scViewerWindow.isEnableInteraction) return
        if(mouseEvent.button == 3) scViewerWindow.dispose()
    }

    override fun mouseReleased(mouseEvent: MouseEvent) {
        super.mouseReleased(mouseEvent)
        if(scViewerWindow.isDefaultImage()) {
            val file = SnipFileChooser.openSystemFileChooser(
                parent = scViewerWindow,
                multiFileSelection = false,
                type = SnipFileChooser.SelectionType.FILES_ONLY,
                fileFilters = listOf(SnipFileChooser.IMAGE_FILTER)
            )?.firstOrNull() ?: return
            scViewerWindow.setImage(file)
        }
    }

    override fun keyPressed(keyEvent: KeyEvent) {
        super.keyPressed(keyEvent)
        if(!scViewerWindow.isEnableInteraction) return
        when (keyEvent.keyCode) {
            KeyEvent.VK_LEFT -> scViewerWindow.slideImage(-1)
            KeyEvent.VK_RIGHT -> scViewerWindow.slideImage(1)
            KeyEvent.VK_ENTER -> scViewerWindow.openEditor()
            KeyEvent.VK_F5 -> scViewerWindow.refreshFolder()
        }
    }
}